diagnostic/engine.go

```diff
@@ -144,112 +236,6 @@ func (c *conflict) addSimilarConflict(conflict conflict) {
 	c.similarConflicts = append(c.similarConflicts, &conflict)
 }
 
-// ConflictList stores a list of conflicts.
-type ConflictList struct {
-	conflicts  []conflict
-	NoGrouping bool // if set to true, conflicts are not grouped by nil path. (Example use case: no-infer unit tests)
-}
-
-// AddSingleAssertionConflict adds a new single assertion conflict to the list of conflicts.
-func (l *ConflictList) AddSingleAssertionConflict(pass *analysis.Pass, trigger annotation.FullTrigger) {
-	t := fullTriggerAsPrimitive(pass, trigger)
-	c := conflict{
-		pos:  t.Pos,
-		flow: nilFlow{},
-	}
-
-	c.flow.addNonNilPathNode(t.ProducerRepr, t.ConsumerRepr)
-
-	l.conflicts = append(l.conflicts, c)
```

Comment:
This is updated a little bit since we do not really need to convert this trigger to primitive trigger to have the necessary information (i.e., the producer and consumer prestrings)